tests/run-pass/exit.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    std::process::exit(0)
```

Comment:
Also add a run-fail test?

Reply:
We only have `compile-fail`... and what would the error pattern look like?

We don't dispatch based on the actual exit code at all, so this seems like a fine test to me.

Reply:
hmm... not sure if compiletest-rs has the comment flags for requiring specific exit codes like rustc's compiletest suite has.

So yea, seems fine this way